tests/test_views.py

```diff
@@ -158,44 +165,194 @@ def test_view_poll_plugin_attached_to_page_should_return_related_page(self):
 
-        # queryset should contain page
-        self.assertEqual(response.context["querysets"][0].count(), 2)
-        self.assertNotIn(archived.content, response.context["querysets"][0])
-        self.assertIn(version2.content, response.context["querysets"][0])
-        self.assertIn(version3.content, response.context["querysets"][0])
+    def test_view_published_filter_applied(self):
+        version1 = PageVersionFactory(
+            content__title="test1", content__language=self.language,
+        )
+        version2 = PageVersionFactory(
+            content__title="test2", content__language=self.language, state=PUBLISHED
+        )
 
-    def test_extra_columns(self):
-        extra_column = ExtraColumn(lambda o: "{} test".format(o), "Test column")
+        page_content1 = version1.content
+        page_content2 = version2.content
 
-        with self.login_user_context(self.superuser), patch(
-            "djangocms_references.views.get_extra_columns", return_value=[extra_column]
-        ):
-            response = self.client.get(self.view_url)
+        placeholder1 = PlaceholderFactory(
+            content_type=ContentType.objects.get_for_model(page_content2),
+            object_id=page_content2.id,
+        )
 
-        self.assertEqual(response.context["extra_columns"], [extra_column])
+        poll1 = PollFactory()
+        # add poll plugin to page
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
+
+        # When published is selected only the published entries should be shown
+        version_selection = f"?state={PUBLISHED}"
+        admin_endpoint = self.get_view_url(
+            content_type_id=ContentType.objects.get_for_model(poll1).pk,
+            object_id=poll1.id,
+        )
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(admin_endpoint + version_selection)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context["querysets"][0].count(), 1)
+        self.assertNotIn(page_content1, response.context["querysets"][0])
+        self.assertIn(page_content2, response.context["querysets"][0])
+
+    def test_view_archived_filter_applied(self):
+        version1 = PageVersionFactory(
+            content__title="test1", content__language=self.language,
+        )
+        version2 = PageVersionFactory(
+            content__title="test2", content__language=self.language, state=ARCHIVED
+        )
+
+        page_content1 = version1.content
+        page_content2 = version2.content
+
+        placeholder1 = PlaceholderFactory(
+            content_type=ContentType.objects.get_for_model(page_content2),
+            object_id=page_content2.id,
+        )
+
+        poll1 = PollFactory()
+        # add poll plugin to page
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
```

Comment:
There's some inconsistencies in the tests. This test will need an additional plugin as per the others.